dawn/src/dawn/CodeGen/Cuda-ico/CudaIcoCodeGen.cpp

```diff
@@ -1204,6 +1204,7 @@ void CudaIcoCodeGen::generateAllCudaKernels(
     std::stringstream& ssSW,
     const std::shared_ptr<iir::StencilInstantiation>& stencilInstantiation) {
   ASTStencilBody stencilBodyCXXVisitor(stencilInstantiation->getMetaData());
+  stencilBodyCXXVisitor.setGenAtlasCompatCode(codeGenOptions_.AtlasCompatible);
```

Comment:
Since we are always setting this, how about adding it to the ctr, instead of having to call a setter ? We might have multiple instances of a ASTStencilBody, and given the compiler flag is like a global option, we might end up with inconsistencies if some are instantiated and call the setter but other dont